### DISCUSSION TITLE
conditions seems to always succeed in flows

### DISCUSSION THREAD

**Discussion by Nisthar** [directus/directus#13851] [Archive]
Condition:

![image](https://user-images.githubusercontent.com/11994157/173224207-950f70f1-df6d-4b30-ab82-4c3b8026b706.png)

flow: 
![image](https://user-images.githubusercontent.com/11994157/173224160-464c772a-9e03-4e16-85bd-b6f6ab61c289.png)

It always goes to the success even if the condition is wrong
![image](https://user-images.githubusercontent.com/11994157/173224182-55352336-9260-4c7d-8961-df6d1d983993.png)

**Comment by rijkvanzanten** [comment]
In this case, you wouldn't use the brackets, as that's used to inject the value of the field you're checking against. In this case, the filter rule should be

```json
{
  "$last": {
    "fghfghfg": {
      "running": {
         "_eq": true
      }
    }
  }
}
```

**Reply by Nisthar** [reply]
> In this case, you wouldn't use the brackets, as that's used to inject the value of the field you're checking against. In this case, the filter rule should be
> 
> ```json
> {
>   "$last": {
>     "fghfghfg": {
>       "running": {
>          "_eq": true
>       }
>     }
>   }
> }
> ```

Its actually an array. i put this value to demonstrate it always succeed. 
Tried it like 
![image](https://user-images.githubusercontent.com/11994157/173425003-99f52ae5-4c7e-46f6-85f3-b52e3a489012.png)

got this error
![image](https://user-images.githubusercontent.com/11994157/173425200-8863c8fc-cc71-40da-84d7-b4f5839d2b7b.png)

I can paste the full log if you want

**Reply by rijkvanzanten** [reply]
Ahh I see. Is the goal to validate against a single item in that array, or all of them? I don't think the current implementation of the condition operation supports validation against a whole array at once 🤔 (cc @nickrum)

**Reply by Nisthar** [reply]
> Ahh I see. Is the goal to validate against a single item in that array, or all of them? I don't think the current implementation of the condition operation supports validation against a whole array at once 🤔 (cc @nickrum)

Only one item in the array. The read operation is giving an array of objects.

**Reply by Nisthar** [reply]
Should i open an issue for this?

**Reply by rijkvanzanten** [reply]
No, I believe this is a new feature to add, not a bug in the current implementation 👍🏻

**Reply by Nisthar** [reply]
Ok i see

**Reply by Nisthar** [reply]
Is there any update on this?

**Reply by rijkvanzanten** [reply]
Nope! Working on it. Keep an eye on the release notes 👍🏻

**Reply by Nisthar** [reply]
> Nope! Working on it. Keep an eye on the release notes 👍🏻

I think this still happens on the latest version of directus. there is no way to use $last inside condition after read data.

**Reply by chris-shepperd** [reply]
I get the same problem. $last doesn't seem to be understood, causes errors whenever used and anything else ($trigger...) I use comes back true regardless of whether the condition is actually true.
